rllib/policy/tf_policy.py

```diff
@@ -1043,70 +1044,38 @@ def _build_compute_actions(
         builder.add_feed_dict(self.extra_compute_action_feed_dict())
 
         # `input_dict` given: Simply build what's in that dict.
-        if input_dict is not None:
-            if hasattr(self, "_input_dict"):
-                for key, value in input_dict.items():
-                    if key in self._input_dict:
-                        # Handle complex/nested spaces as well.
-                        tree.map_structure(
-                            lambda k, v: builder.add_feed_dict({k: v}),
-                            self._input_dict[key],
-                            value,
-                        )
-            # For policies that inherit directly from TFPolicy.
-            else:
-                builder.add_feed_dict({self._obs_input: input_dict[SampleBatch.OBS]})
-                if SampleBatch.PREV_ACTIONS in input_dict:
-                    builder.add_feed_dict(
-                        {self._prev_action_input: input_dict[SampleBatch.PREV_ACTIONS]}
-                    )
-                if SampleBatch.PREV_REWARDS in input_dict:
-                    builder.add_feed_dict(
-                        {self._prev_reward_input: input_dict[SampleBatch.PREV_REWARDS]}
+        if hasattr(self, "_input_dict"):
```

Comment:
It's soft-deprecated since the trajectory view API was introduced. You shouldn't call this method anymore with anything other than an `input_dict`, such that you have flexibility as to the different field names, e.g. you could have a "prev_10_actions" in your input_dict (was not supported before due to hardcoded `prev_action|reward` schema).

Reply:
wow, ok, should have been deprecated long time ago then.
cool.